src/main/java/teammates/ui/webapi/UpdateFeedbackSessionAction.java

```diff
@@ -103,21 +108,93 @@ public JsonResult execute() throws InvalidHttpRequestBodyException {
                             .withStudentDeadlines(studentDeadlines)
                             .withInstructorDeadlines(instructorDeadlines)
                             .build());
-
-            if (!studentDeadlines.equals(oldStudentDeadlines) || !instructorDeadlines.equals(oldInstructorDeadlines)) {
-                taskQueuer.scheduleChangesToDeadlineExtensions(courseId, feedbackSessionName,
-                        notifyAboutDeadlines, oldStudentDeadlines, studentDeadlines, oldInstructorDeadlines,
-                        instructorDeadlines);
-            }
-
-            return new JsonResult(new FeedbackSessionData(updateFeedbackSession));
         } catch (InvalidParametersException ipe) {
             throw new InvalidHttpRequestBodyException(ipe);
         } catch (EntityDoesNotExistException ednee) {
             // Entity existence has been verified before, and this exception should not happen
             log.severe("Unexpected error", ednee);
             return new JsonResult(ednee.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
         }
+
+        boolean notifyAboutDeadlines = getBooleanRequestParamValue(Const.ParamsNames.NOTIFY_ABOUT_DEADLINES);
+
+        List<EmailWrapper> emailsToSend = new ArrayList<>();
+        if (!studentDeadlines.equals(oldStudentDeadlines)) {
+            processDeadlineExtensions(courseId, feedbackSession, oldStudentDeadlines, studentDeadlines,
+                    false, notifyAboutDeadlines, emailsToSend);
+        }
+
+        if (!instructorDeadlines.equals(oldInstructorDeadlines)) {
+            processDeadlineExtensions(courseId, feedbackSession, oldInstructorDeadlines, instructorDeadlines,
+                    true, notifyAboutDeadlines, emailsToSend);
+        }
```

Comment:
Would it be cleaner to put the condition inside the `processDeadlineExtensions` method, so that other future user of this method will not need to take note of making the same `if` block

Reply:
sounds like a good idea!